### ISSUE TITLE
`REFRESH NEVER` materialized views

### ISSUE THREAD

**Issue by hlburak** [MaterializeInc/materialize#23012]
# Product Outcome

TLDR: Users can reduce their costs of querying cold data in Materialize by creating a materialized view which never refreshes.

[Copied from tuning freshness design doc](https://github.com/ggevay/materialize/blob/refresh-every-design-doc/doc/developer/design/20231027_refresh_every_mv.md)

ODW use cases typically focus on the most recent input data (hot data). However, there is often a significantly larger amount of older data lying around (cold data). The cold data is often of the same kind as the hot data, and therefore we postulate that users will often want to perform the same or similar computations on these. In this case, it would be convenient for users to simply use Materialize for both hot and cold computations, and not complicate their architectures by implementing the same computations in both Materialize and a different, traditional data warehouse.

The problem is that Materialize is currently not so cost-effective for running a materialized view on cold data, because

- there is significantly more cold data than hot data, so a much larger compute instance is needed to keep compute state in memory, and
- keeping results always fresh is not so valuable for cold data.

This currently prevents us from capturing these cold use cases, even though they feel very close to our hot, core ODW use cases.

# Discovery

Request described well [here](https://materializeinc.slack.com/archives/C03REC5MEKU/p1698928179127449).

**Comment by hlburak** [comment]
This work will be delivered with `REFRESH EVERY...`

**Comment by ggevay** [comment]
I've turned this into a normal (non-epic) issue and added it to the `REFRESH OPTIONS` epic as a task.

**Comment by ggevay** [comment]
This is `REFRESH AT` currently.
